tests/interpreters/test_affine_interpreter.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any
+
+from xdsl.builder import Builder, ImplicitBuilder
+from xdsl.dialects import affine, arith, func, memref
+from xdsl.dialects.builtin import IndexType, ModuleOp
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.affine import AffineFunctions
+from xdsl.interpreters.arith import ArithFunctions
+from xdsl.interpreters.func import FuncFunctions
+from xdsl.interpreters.memref import MemrefFunctions
+from xdsl.interpreters.shaped_array import ShapedArray
+
+index = IndexType()
+
+
+@ModuleOp
+@Builder.implicit_region
+def module_op():
+    with ImplicitBuilder(
+        func.FuncOp(
+            "my_func",
+            ((), (memref.MemRefType.from_element_type_and_shape(index, (2, 3)),)),
+        ).body
+    ):
+        alloc_op = memref.Alloc.get(index, None, (2, 3))
+
+        @Builder.implicit_region((index,))
+        def rows_region(args: tuple[Any, ...]):
+            (row,) = args
+
+            @Builder.implicit_region((index,))
+            def cols_region(args: tuple[Any, ...]):
+                (col,) = args
+                sum_op = arith.Addi(row, col)
+                affine.Store(sum_op.result, alloc_op.memref, (row, col))
```

Comment:
Can you add a Load in the exemple?  NB I didn't foloow the memref implementation closely yet, not sure how hard this would be!

Reply:
Ah good point forgot about that.
- [ ] add load